src/helpers/git.js

```diff
@@ -97,7 +97,7 @@ module.exports = new (class Git {
    * @return {Promise<>}
    */
   pull = () => (
-    this.exec(`pull --unshallow ${core.getInput('git-pull-method')}`)
+    this.exec(`pull ${core.getInput('git-pull-method')}`)
```

Comment:
Is there a reason why you removed the unshallow? That should also make sure all the git tags are retrieved.

Reply:
i have created a test repository with a workflow that runs `git log --decorate --no-color`

```yaml
- name: Checkout
  uses: actions/checkout@v2
  with:
    fetch-depth: 0
- name: Pull
  run: |
    git pull --unshallow --ff-only
```
--> `fatal: --unshallow on a complete repository does not make sense`
```yaml
steps:
  - name: Checkout
    uses: actions/checkout@v2
  - name: Pull
    run: |
      git pull --unshallow --ff-only
```
--> no tags available in log
... `commit f5d3880d5518d39253e3a0f59c07bc13c824a762` ...

```yaml
steps:
  - name: Checkout
    uses: actions/checkout@v2
    with:
      fetch-depth: 0
  - name: Pull
    run: |
      git pull --ff-only
```
--> tags available in log 
... `commit f5d3880d5518d39253e3a0f59c07bc13c824a762 (tag: 0.1.0)` ...

Reply:
Thanks for the explanation, personally I would not remove it as doing so and requiring uses to update their checkout action will result in a breaking change. Think its better for this pr to only do the configurable fallback version and pheraps change this in v4.

Reply:
found another solution:
`git pull --unshallow --tags --ff-only` --> ... `commit f5d3880d5518d39253e3a0f59c07bc13c824a762 (tag: 0.1.0)` ...